Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -40,6 +47,131 @@ class Transition(NamedTuple):
+    # The 5 Native Games are currently chosen to not be randomized.
+    # If we at some point decide to randomize them anyway we'll have to do some rigorous testing
+    # Because it's very much possible this will cause some bugs
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.WHACK_A_TUCO),
+    (LevelCRC.WHACK_A_TUCO, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.TUCO_SHOOT),
+    (LevelCRC.TUCO_SHOOT, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.RAFT_BOWLING),
+    (LevelCRC.RAFT_BOWLING, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.PICKAXE_RACE),
+    (LevelCRC.PICKAXE_RACE, LevelCRC.NATIVE_VILLAGE),
+    (LevelCRC.NATIVE_VILLAGE, LevelCRC.KABOOM),
+    (LevelCRC.KABOOM, LevelCRC.NATIVE_VILLAGE),
+    # The 2 CUTSCENE Levels are currently chosen to not be randomized.
+    # As of right now both of these cutscenes are hijacked to be skipped entirely
+    (LevelCRC.JAGUAR, LevelCRC.PLANE_CUTSCENE),
+    (LevelCRC.PLANE_CUTSCENE, LevelCRC.CRASH_SITE),
+    (LevelCRC.SPINJA_LAIR, LevelCRC.VIRACOCHA_MONOLITHS_CUTSCENE),
+    (LevelCRC.VIRACOCHA_MONOLITHS_CUTSCENE, LevelCRC.VIRACOCHA_MONOLITHS),
+    # Specific one-time, one-way warps are not randomized.
+    # These 3 are handled elsewhere in the randomizer
+    # Currently there are no plans to randomize these transitions.
+    (LevelCRC.ALTAR_OF_AGES, LevelCRC.BITTENBINDERS_CAMP),
+    (LevelCRC.ST_CLAIRE_DAY, LevelCRC.ST_CLAIRE_NIGHT),
+    (LevelCRC.ST_CLAIRE_NIGHT, LevelCRC.ST_CLAIRE_DAY),
+    # The Endgame El Dorado transitions are not randomized.
+    # Currently there are no plans to randomize these transitions.
+    (LevelCRC.GATES_OF_EL_DORADO, LevelCRC.JAGUAR),
+    (LevelCRC.JAGUAR, LevelCRC.PUSCA),
+    (LevelCRC.PUSCA, LevelCRC.GATES_OF_EL_DORADO),
+    # The Unused Beta Volcano Level is not randomized yet,
+    # but this can absolutely be randomized later at some point.
+    # It will require some special code though.
+    (LevelCRC.BETA_VOLCANO, LevelCRC.JUNGLE_CANYON),
+    (LevelCRC.BETA_VOLCANO, LevelCRC.PLANE_COCKPIT),
+]
+
+TRANSITION_INFOS_DICT_RANDO = TRANSITION_INFOS_DICT.copy()
+ALL_POSSIBLE_TRANSITIONS_RANDO = ALL_POSSIBLE_TRANSITIONS
+
+
+def remove_disabled_exits():
+    # remove exits from TRANSITION_INFOS_DICT_RANDO
+    for area in TRANSITION_INFOS_DICT.values():
+        for ex in area.exits:
+            current = (area.area_id, ex.area_id)
+            if current in one_way_exits or current in disabled_exits:
+                TRANSITION_INFOS_DICT_RANDO[area.area_id].exits = (
+                    x for x in TRANSITION_INFOS_DICT_RANDO[area.area_id].exits if x != ex
+                )
+                area.con_left -= 1
+
+    # remove exits from ALL_POSSIBLE_TRANSITIONS_RANDO
+    global ALL_POSSIBLE_TRANSITIONS_RANDO
+    for trans in ALL_POSSIBLE_TRANSITIONS:
+        if trans in one_way_exits or trans in disabled_exits:
+            ALL_POSSIBLE_TRANSITIONS_RANDO = [
```

Comment:
I'm sure there's a cleaner way to handle this than overriding global constants. But at least now this code is self-contained for the entrance rando and this is the code responsible for resetting `ALL_POSSIBLE_TRANSITIONS_RANDO` if we, for example, allow starting a new seed w/o having to reload the script.

```suggestion
            ALL_POSSIBLE_TRANSITIONS_RANDO = [  # pyright: ignore[reportConstantRedefinition]
```

Reply:
comment is added.